TheOracle2/Commands/PlayerRollCommand.cs

```diff
@@ -16,42 +17,84 @@ public PlayerRollCommand(Random random, EFContext efContext)
 
     public Random Random { get; }
     public EFContext EfContext { get; }
+    public GuildPlayer GuildPlayer => GuildPlayer.AddIfMissing(Context, EfContext);
+
+    public override async Task AfterExecuteAsync(ICommandInfo command)
+    {
+        await EfContext.SaveChangesAsync().ConfigureAwait(false);
+    }
 
     [SlashCommand("action-pc-roll", "Performs an Ironsworn action roll using a player character's stats.")]
     public async Task ActionRoll(
-        [Summary(description: "The character to use for the roll")][Autocomplete(typeof(CharacterAutocomplete))] string character,
         [Summary(description: "The stat value to use for the roll")] RollableStats stat,
         [Summary(description: "Any adds to the roll")][MinValue(0)] int adds,
+        [Summary(description: "The character to use for the roll. Leave this blank to use the last PC you interacted with.")]
+        // -1 is used to represent "last used character"
+        [Autocomplete(typeof(CharacterAutocomplete))] string character = "-1",
         [Summary(description: "Any notes, fiction, or other text you'd like to include with the roll")] string description = "",
         [Summary(description: "A preset value for the Action Die (d6) to use instead of rolling.")][MinValue(1)][MaxValue(6)] int? actionDie = null,
         [Summary(description: "A preset value for the first Challenge Die (d10) to use instead of rolling.")][MinValue(1)][MaxValue(10)] int? challengeDie1 = null,
         [Summary(description: "A preset value for the second Challenge Die (d10) to use instead of rolling.")][MinValue(1)][MaxValue(10)] int? challengeDie2 = null)
     {
-        if (!int.TryParse(character, out int id))
+        var id = 0;
+        PlayerCharacter pcData = null;
+
+        if (!int.TryParse(character, out id))
         {
-            await RespondAsync($"Unknown character", ephemeral: true);
-            return;
+            throw new Exception();
```

Comment:
i've changed the ones here to ArgumentExceptions, but there's others lurking elsewhere. because of the sheer amount of files that get moved around in #29, i'm going to handle the other exceptions in separate PR.